src/renderer/api/kube-api.ts

```diff
@@ -86,15 +93,35 @@ export class KubeApi<T extends KubeObject = any> {
     this.apiPrefix = apiPrefix;
     this.apiGroup = apiGroup;
     this.apiVersion = apiVersion;
-    this.apiVersionWithGroup = apiVersionWithGroup;
     this.apiResource = resource;
     this.request = request;
     this.objectConstructor = objectConstructor;
 
+    this.checkPreferredVersion();
     this.parseResponse = this.parseResponse.bind(this);
     apiManager.registerApi(apiBase, this);
   }
 
+  get apiVersionWithGroup() {
+    return [this.apiGroup, this.apiVersionPreferred ?? this.apiVersion]
+      .filter(Boolean)
+      .join("/")
+  }
+
+  protected async checkPreferredVersion() {
+    if (this.options.checkPreferredVersion && this.apiVersionPreferred === undefined) {
+      const res = await this.request.get<IKubePreferredVersion>(`${this.apiPrefix}/${this.apiGroup}`);
+      Object.defineProperty(this, "apiVersionPreferred", {
+        value: res?.preferredVersion?.version ?? null,
+      });
```

Comment:
Just looking through PRs to know what has changed.

May I ask why this is required instead of `this.apiVersionPreferred = res?.preferredVersion?.version ?? null;`?